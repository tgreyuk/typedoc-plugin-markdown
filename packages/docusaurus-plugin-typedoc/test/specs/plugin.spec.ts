import * as fs from 'fs-extra';
import * as tmp from 'tmp';
import typedocPlugin from '../../dist/plugin';

tmp.setGracefulCleanup();

async function bootstrap(tmpobj: tmp.DirResult, customOptions = {}) {
  const options = {
    id: 'default',
    // logger: 'none',
    entryPoints: ['../../stubs/src/theme.ts'],
    tsconfig: '../../stubs/tsconfig.json',
  } as any;

  const plugin = typedocPlugin(
    {
      siteDir: tmpobj.name,
      generatedFilesDir: '',
      siteConfig: {},
    } as any,
    {
      ...options,
      ...customOptions,
    },
  );
  return await plugin.loadContent();
}

describe(`Plugin:`, () => {
  describe(`(docs)`, () => {
    let tmpobj;
    beforeAll(async () => {
      tmpobj = tmp.dirSync();
      await bootstrap(tmpobj);
    });

    test(`should write doc`, () => {
      const indexDoc = fs.readFileSync(tmpobj.name + '/docs/api/index.md');
      expect(indexDoc.toString()).toMatchSnapshot();
    });

    test(`should write category yaml`, () => {
      const categoryYaml = fs.readFileSync(
        tmpobj.name + '/docs/api/classes/_category_.yml',
      );
      expect(categoryYaml.toString()).toMatchSnapshot();
    });
  });
});
