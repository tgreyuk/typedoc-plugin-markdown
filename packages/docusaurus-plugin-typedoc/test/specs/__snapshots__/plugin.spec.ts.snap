// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin: (docs) should write category yaml 1`] = `
"label: "Classes"
position: 3"
`;

exports[`Plugin: (docs) should write doc 1`] = `
"---
sidebar_label: "Readme"
sidebar_position: 0
custom_edit_url: null
---

# MY README

[API](Modules.md)

> Some block quote

<!-- some comments -->

Some text
"
`;
